[SwiftASTContext] Share common code with PlatformDarwin.

JDevlieghere opened the pull request:
Commit 7aa28995e87f6cec2e02608c978885cf01dcc65d refactored the Xcode
resource directory logic and made it possible to have this code shared
with SwiftASTContext.

JDevlieghere commented:
@swift-ci please test

JDevlieghere commented:
I'll cherry-pick this to `master-next` once this is approved and landed.

adrian-prantl commented on lldb/source/Symbol/SwiftASTContext.cpp:
```diff
@@ -1536,14 +1536,18 @@ static StringRef GetXcodeContentsPath() {
+          // Strip any leading newline characters and everything before them.
+          // This needs to happen *after* the erase above because otherwise a
+          // trailing newline might erase the whole string.
+          size_t last_newline = output.find_last_of("\r\n");
```
I don't see why the `\r` is necessary here. We are not going to run this on macOS < 10.0. On Windows `\r\n` will still contain `\n`.

adrian-prantl replied:
Shouldn't/Isn't this function in lldb/source/Plugins/Platform/MacOSX/PlatformMacOSX.cpp?

adrian-prantl replied:
ConstString PlatformMacOSX::GetSDKDirectory(lldb_private::Target &target) {

JDevlieghere replied:
Good point, this function is doing something different but it's also calling `xcrun`. Let me factor that out and reuse it in both places.

dcci reviewed (commented):
This is really interesting, thanks. Is this commit motivated by a bug you hit while parsing the SDK or it's just a drive-by fix?

JDevlieghere commented:
> This is really interesting, thanks. Is this commit motivated by a bug you hit while parsing the SDK or it's just a drive-by fix?

Yup, it was what caused my REPL tests to fail. Apparently `fish` is printing some warning before the path now that we honor the `$SHELL` variable. That also explains why they still failed when I ran the test under a different shell.

dcci reviewed (commented):
LGTM, but Adrian should sign-off too.

dcci commented:
@swift-ci test

JDevlieghere commented:
This depends on an upstream change that hasn't landed yet. I think we can remove even more complexity so I'm going to rework the upstream patch a bit.

JDevlieghere committed:
[SwiftASTContext] Share common code with PlatformDarwin.

Commit 7aa28995e87f6cec2e02608c978885cf01dcc65d refactored the Xcode
resource directory logic and made it possible to have this code shared
with SwiftASTContext.

JDevlieghere commented:
Depends on #921

JDevlieghere commented:
@swift-ci please test

adrian-prantl reviewed (approved):
This is clearly an improvement, but really these functions should all be in PlatfromDarwin on llvm.org.

JDevlieghere commented:
The AM claimed that there was nothing to merge from 20200108 into swift/master but the Linux bot says the dependency isn't there...

JDevlieghere commented:
@swift-ci please test

JDevlieghere committed:
[SwiftASTContext] Use llvm::sys::path for path manipulation

Manipulating paths as strings is error prone. This rewrites the SDK
findings logic in terms of llvm::sys::path.

JDevlieghere commented:
@swift-ci please test
